#==================================================================================================
#
# Name: lldb - Get rid of __STDC_LIMIT_MACROS and __STDC_CONSTANT_MACROS
# Date: 2022-09-29
# Link: https://reviews.llvm.org/D134877
#
# Summary:
#   C++11 made the use of these macro obsolete, see:
#   https://sourceware.org/bugzilla/show_bug.cgi?id=15366
#
# Trac Ticket: https://trac.macports.org/ticket/67828
#
#==================================================================================================
--- a/tools/lldb/bindings/CMakeLists.txt
+++ b/tools/lldb/bindings/CMakeLists.txt
@@ -26,8 +26,6 @@
   -features autodoc
   -I${LLDB_SOURCE_DIR}/include
   -I${CMAKE_CURRENT_SOURCE_DIR}
-  -D__STDC_LIMIT_MACROS
-  -D__STDC_CONSTANT_MACROS
   ${DARWIN_EXTRAS}
 )
 
--- a/tools/lldb/bindings/interfaces.swig
+++ b/tools/lldb/bindings/interfaces.swig
@@ -1,8 +1,5 @@
 /* Various liblldb typedefs that SWIG needs to know about.  */
 #define __extension__ /* Undefine GCC keyword to make Swig happy when processing glibc's stdint.h. */
-/* The ISO C99 standard specifies that in C++ implementations limit macros such
-   as INT32_MAX should only be defined if __STDC_LIMIT_MACROS is. */
-#define __STDC_LIMIT_MACROS
 %include "stdint.i"
 
 %include "lldb/lldb-defines.h"

